File: src/main/java/no/mesan/fag/patterns/timesheet/external/TimeDataServer.java
package no.mesan.fag.patterns.timesheet.external;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import no.mesan.fag.patterns.timesheet.data.TimesheetEntry;

/**
 * Implementasjon av TimeDataService.
 */
public class TimeDataServer implements TimeDataService {

    private final TimeSource source;

    /** Default constructor. */
    public TimeDataServer() {
        this.source = new TimeSource();
    }

    @Override
    public List<TimesheetEntry> forEmployee(final String userID) {
        return StreamSupport.stream(source.spliterator(), false)
                .filter(entry -> entry.getUserID().equals(userID))
                .collect(Collectors.toList());
    }
}

File: src/main/scala/no/mesan/fag/patterns/scala/timesheet/external/TimeDataServer.scala
package no.mesan.fag.patterns.scala.timesheet.external

import scala.language.postfixOps
import no.mesan.fag.patterns.scala.timesheet.data.TimesheetEntry

/** Dette er liksom selve den eksterne serveren. Tilgi meg, hadde ikke råd til en stormaskin til hver. */
class TimeDataServer(src: Iterable[TimesheetEntry]) extends TimeDataService {

  override def forEmployee(userID: String, from: Int): List[TimesheetEntry] =
    (src filter (_.userID == userID)).slice(from, from + BatchSize) toList

  override def forYear(year: Int, from: Int): List[TimesheetEntry] =
    (src filter (_.when.getYear == year)).slice(from, from + BatchSize) toList
}
